File: BackupUtility/Classes/Readablity/BasicReadChecker.vb
Public Class BasicReadChecker
    Implements IFileReadableChecker

    Public Function CanReadFile(ByVal aFile As String) As Boolean Implements IFileReadableChecker.CanReadFile
        'Try if we can read the file
        If aFile Is Nothing OrElse aFile = "" Then Return False
        Return True
    End Function

End Class

File: BackupUtility/Classes/Readablity/BasicReadChecker.cs
using ZCopy.Interfaces;

namespace ZCopy.Classes.Readability
{
    public class BasicReadChecker : IFileReadableChecker
    {
        public bool CanReadFile(string aFile)
        {
            // Try if we can read the file
            if (string.IsNullOrEmpty(aFile))
                return false;
            return true;
        }
    }
}
